File: src/main/java/io/realworld/api/CommentsApi.java
package io.realworld.api;

import io.realworld.model.CreateArticleComment200Response;
import io.realworld.model.CreateArticleCommentRequest;
import io.realworld.model.GetArticleComments200Response;

import io.vertx.core.Future;

public interface CommentsApi  {
    Future<ApiResponse<CreateArticleComment200Response>> createArticleComment(String slug, CreateArticleCommentRequest comment);
    Future<ApiResponse<Void>> deleteArticleComment(String slug, Integer id);
    Future<ApiResponse<GetArticleComments200Response>> getArticleComments(String slug);
}

File: src/main/kotlin/io/realworld/api/CommentsApi.kt
package io.realworld.api

import io.realworld.model.request.CreateArticleCommentRequest
import io.realworld.model.response.ApiResponse
import io.realworld.model.response.CreateArticleComment200Response
import io.realworld.model.response.GetArticleComments200Response
import io.vertx.core.Future

interface CommentsApi {
  fun createArticleComment(
    slug: String,
    comment: CreateArticleCommentRequest
  ): Future<ApiResponse<CreateArticleComment200Response>>

  fun deleteArticleComment(slug: String, id: Int): Future<ApiResponse<Void>>
  fun getArticleComments(slug: String): Future<ApiResponse<GetArticleComments200Response>>
}
